lib/fail.js

```diff
@@ -0,0 +1,60 @@
+const {template} = require('lodash');
+const urlJoin = require('url-join');
+const got = require('got');
+const debug = require('debug')('semantic-release:gitlab');
+const resolveConfig = require('./resolve-config');
+const getRepoId = require('./get-repo-id');
+const getFailComment = require('./get-fail-comment');
+
+module.exports = async (pluginConfig, context) => {
+  const {
+    options: {repositoryUrl},
+    branch,
+    errors,
+    logger,
+  } = context;
+  const {gitlabToken, gitlabUrl, gitlabApiUrl, failComment, failTitle, labels, assignee} = resolveConfig(
+    pluginConfig,
+    context
+  );
+  const repoId = getRepoId(context, gitlabUrl, repositoryUrl);
+  const encodedRepoId = encodeURIComponent(repoId);
+  const apiOptions = {headers: {'PRIVATE-TOKEN': gitlabToken}};
+
+  if (failComment === false || failTitle === false) {
+    logger.log('Skip issue creation.');
+  } else {
+    const description = failComment ? template(failComment)({branch, errors}) : getFailComment(branch, errors);
+
+    const issuesEndpoint = urlJoin(gitlabApiUrl, `/projects/${repoId}/issues`);
+    const openIssueEndpoint = urlJoin(issuesEndpoint, '?state=opened&per_page=100');
```

Comment:
We have projects with a huge amount of issues. The Gitlab API returns by default only the [first 20 issues](https://docs.gitlab.com/ee/api/#offset-based-pagination). This is why I increased the `per_page` size to the `max` of `100`.
Is this OK or should we use the default size. Or should we even search through all pages?

(Might also be relevant for the `success`-hook)

Reply:
Oh, that's a good finding! 

I don't like the `100` either because it just defers the problem.

What about using the `search` parameter to prefilter for the expected title: https://docs.gitlab.com/ee/api/issues.html?msclkid=31d7fcfdb4b211ec99728d2bf9d701dc#list-project-issues